extra/resources/CPUIdleHealth

```diff
@@ -0,0 +1,192 @@
+the #health-cpu-idle will go red if the %idle of the CPU falls below 15%.
+</longdesc>
+<shortdesc lang="en">Lower limit for red health attribute</shortdesc>
+<content type="string" default="15"/>
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start"        timeout="10s" />
+<action name="stop"         timeout="10s" />
+<action name="monitor"      timeout="10s" interval="10s" start-delay="0s" />
+<action name="meta-data"    timeout="5s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+dummy_usage() {
+	cat <<END
+usage: $0 {start|stop|monitor|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+dummy_start() {
+    dummy_monitor
+    if [ $? =  $OCF_SUCCESS ]; then
+	return $OCF_SUCCESS
+    fi
+    touch ${OCF_RESKEY_state}
+}
+
+dummy_stop() {
+    dummy_monitor
+    if [ $? =  $OCF_SUCCESS ]; then
+	rm ${OCF_RESKEY_state}
+    fi
+    return $OCF_SUCCESS
+}
+
+dummy_monitor() {
+	# Monitor _MUST!_ differentiate correctly between running
+	# (SUCCESS), failed (ERROR) or _cleanly_ stopped (NOT RUNNING).
+	# That is THREE states, not just yes/no.
+	
+	if [ -f ${OCF_RESKEY_state} ]; then
+
+	  IDLE=`iostat -c -y 1 1 | awk '/^ /{printf "%d\r\n", $6}'`
+	  # echo "System idle: " $IDLE
+	  # echo "$OCF_RESKEY_red_limit"
+	  # echo $OCF_RESKEY_yellow_limit
+
+	  if [ $IDLE -lt ${OCF_RESKEY_red_limit} ] ; then
+	    # echo "System state RED!"
+	    attrd_updater -n "#health-cpu-idle" -U "red" -d "30s"
+	    return $OCF_SUCCESS
+          fi
+
+	  if [ $IDLE -lt ${OCF_RESKEY_yellow_limit} ] ; then
+	    # echo "System state yellow."
+            attrd_updater -n "#health-cpu-idle" -U "yellow" -d "30s"
+	  else
+	    # echo "System state green."
+	    attrd_updater -n "#health-cpu-idle" -U "green" -d "30s"
+          fi
+	
+	  return $OCF_SUCCESS
+	fi
+
+	if false ; then
+                return $OCF_ERR_GENERIC
+        fi
+        return $OCF_NOT_RUNNING
+}
+
+dummy_validate() {
+    
+    # Is the state directory writable? 
+    state_dir=`dirname "$OCF_RESKEY_state"`
+    touch "$state_dir/$$"
+    if [ $? != 0 ]; then
+	return $OCF_ERR_ARGS
+    fi
+    rm "$state_dir/$$"
+
+    return $OCF_SUCCESS
+}
+
+: ${OCF_RESKEY_CRM_meta_interval=0}
+: ${OCF_RESKEY_CRM_meta_globally_unique:="false"}
+
+if [ "x$OCF_RESKEY_state" = "x" ]; then
+    if [ ${OCF_RESKEY_CRM_meta_globally_unique} = "false" ]; then
+	state="${HA_VARRUN%%/}/Dummy-${OCF_RESOURCE_INSTANCE}.state"
+	
+	# Strip off the trailing clone marker
+	OCF_RESKEY_state=`echo $state | sed s/:[0-9][0-9]*\.state/.state/`
+    else 
+	OCF_RESKEY_state="${HA_VARRUN%%/}/Dummy-${OCF_RESOURCE_INSTANCE}.state"
+    fi
+fi
+
+if [ "x${OCF_RESKEY_red_limit}" = "x" ] ; then
+  OCF_RESKEY_red_limit=15
+fi
+
+if [ "x${OCF_RESKEY_yellow_limit}" = "x" ] ; then
+  OCF_RESKEY_yellow_limit=50
+fi
```

Comment:
Your defaults here don't match what was documented in the meta-data. Also, with red limit higher than yellow limit, the result will never be yellow, but straight to red.

Reply:
With this plugin - the more IO the worse the score gets.

I took a look and it seems that agent_monitor() works OK. The load of 15 will only fire the rule for RED and exit immiediately. Load of 10 would match yellow only. I will test the case of setting yellow just to make sure on my cluster.

In any case I am about to fix that default limit.

Reply:
The plugin was tested, indeed there was a bug, but it was due to wrong variable name usage.

Yellow threshold is set as intended - tested on my dev cluster. You can double verify it.